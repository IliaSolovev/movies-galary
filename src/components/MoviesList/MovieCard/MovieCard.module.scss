@import "../../../styles/variables";
.movieCard{
  width: 250px;
  height: 400px;
  font-family: $mainFont;
  border: 1px solid rgba(207, 207, 207, 0.24);
  &__poster{
    width: 100%;
    height: 82%;
    margin-bottom: 4%;
    img{
      width: 100%;
      height: 100%;
    }
  }
  &__description{
    width: 100%;
    height: 14%;
    padding: 0 5px;
    display: flex;
    justify-content: space-between;
    box-sizing: border-box;
  }
  &__titleAndGenre{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
  &__title{
    font-weight: bold;
    font-size: .9em;
    color: rgba(21, 21, 21, 0.7);
  }
  &__genre{
    font-weight: normal;
    font-size: .8em;
    color: rgba(0, 0, 0, 0.60);
  }
  &__year{
    width: 40px;
    height: 15px;
    border: 1px solid #b3b3b3;
    border-radius: 2px;
    font-size: .8em;
    font-weight: bold;
    color: #939393;
    text-align: center;
  }
}
