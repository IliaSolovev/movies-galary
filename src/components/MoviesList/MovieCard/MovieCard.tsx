import React from "react";
import s from "./movieCard.module.scss";
import {Movie} from "../../../redux/moviesSlice";

interface Props {
    data: Movie
}

const MovieCard: React.FC<Props> = ({data, ...rest}: Props) => {
    return (
        <div className={s.movieCard}>
                <div className={s.movieCard__poster}>
                    <img src={data.poster_path} alt={data.title}/>
                </div>

            <div className={s.movieCard__description}>
                <div className={s.description__titleAndGenre}>
                    <div className={s.description__title}>{data.title}</div>
                    <div className={s.description__genre}>{data.genres.join(' & ')}</div>
                </div>
                <div className={s.description__year}>{data.release_date.split('-')[0]}</div>
            </div>
        </div>
    )
};

export default MovieCard