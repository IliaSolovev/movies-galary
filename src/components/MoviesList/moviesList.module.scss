@import "../../styles/main";
.movieList {
  @extend .container;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  padding: 15px 0;
  @media screen and(max-width: 640px) {
    justify-content: space-around;
  }
  & > * {
    margin: 10px;
  }
}