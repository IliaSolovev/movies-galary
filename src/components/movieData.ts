import { MovieCardData, MovieDescriptionData } from '../types';

export const movieCard2: MovieCardData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  release_date: '2018-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  genres: [
    'Drama',
    'Romance',
  ],
};
export const movieCard1: MovieCardData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  release_date: '2019-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  genres: [
    'Drama',
    'Romance',
  ],
};

export const movieDescription: MovieDescriptionData = {
  id: '337167',
  title: 'Fifty Shades Freed',
  tagline: "Don't miss the climax",
  rating: 6.1,
  release_date: '2019-02-07',
  poster_path: 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
  overview: 'Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable '
    + 'connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, '
    + 'new threats could jeopardize their happy ending before it even begins.',
  genres: [
    'Drama',
    'Romance',
  ],
  runtime: 106,
};
