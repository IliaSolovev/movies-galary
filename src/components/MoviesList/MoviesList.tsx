import React from 'react';
import { Link } from 'react-router-dom';

import { MovieCard } from './MovieCard/MovieCard';
import { SortFunctions } from '../../services/SortMoviesList';
import { FilmsNotFound } from '..';
import { Movie } from '../../redux';
import { Filters } from '../../redux/moviesReducer';

import style from './moviesList.module.scss';


interface Props {
    movies: Movie[],
    sortFilter: Filters
}

export const MoviesList: React.FC<Props> = ({ movies = [], sortFilter }) => {
  const sortFn = sortFilter === 'rating' ? SortFunctions.rating : SortFunctions.releaseDate;

  const moviesCard = [...movies].sort(sortFn)
    .map((movie) => <Link to={`/movie/${movie.id}`} key={movie.id}><MovieCard movie={movie} /></Link>);

  return (
    <div className={style.movieList}>
      { movies.length === 0 ? <FilmsNotFound /> : moviesCard }
    </div>
  );
};
