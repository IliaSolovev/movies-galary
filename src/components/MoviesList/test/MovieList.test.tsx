import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MovieCard } from '../MovieCard/MovieCard';
import { MoviesList } from '../MoviesList';
import { movieCard2 } from '../../movieData';

it('should render movie card', () => {
  const component = shallow(<MovieCard movie={movieCard2} />);
  expect(toJson(component)).toMatchSnapshot();
});
it('should render movie list', () => {
  const component = shallow(<MoviesList sortFilter="rating" movies={[movieCard2, movieCard2]} />);
  expect(toJson(component)).toMatchSnapshot();
});
it('should render 2 movie', () => {

});
