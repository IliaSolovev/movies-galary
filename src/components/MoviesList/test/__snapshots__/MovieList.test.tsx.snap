// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render movie card 1`] = `
<div
  className="movieCard"
>
  <div
    className="movieCard__poster"
  >
    <img
      alt="Fifty Shades Freed"
      src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
    />
  </div>
  <div
    className="movieCard__description"
  >
    <div
      className="description__titleAndGenre"
    >
      <div
        className="description__title"
      >
        Fifty Shades Freed
      </div>
      <div
        className="description__genre"
      >
        Drama & Romance
      </div>
    </div>
    <div
      className="description__year"
    >
      2018
    </div>
  </div>
</div>
`;

exports[`should render movie list 1`] = `
<Fragment>
  <div
    className="movieList"
  >
    <Link
      key="337167"
      to="/movie/337167"
    >
      <MovieCard
        data={
          Object {
            "budget": 55000000,
            "genres": Array [
              "Drama",
              "Romance",
            ],
            "id": 337167,
            "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
            "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
            "release_date": "2018-02-07",
            "revenue": 136906000,
            "runtime": 106,
            "tagline": "Don't miss the climax",
            "title": "Fifty Shades Freed",
            "vote_average": 6.1,
            "vote_count": 1195,
          }
        }
      />
    </Link>
    <Link
      key="337167"
      to="/movie/337167"
    >
      <MovieCard
        data={
          Object {
            "budget": 55000000,
            "genres": Array [
              "Drama",
              "Romance",
            ],
            "id": 337167,
            "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
            "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
            "release_date": "2018-02-07",
            "revenue": 136906000,
            "runtime": 106,
            "tagline": "Don't miss the climax",
            "title": "Fifty Shades Freed",
            "vote_average": 6.1,
            "vote_count": 1195,
          }
        }
      />
    </Link>
  </div>
</Fragment>
`;
